core/server/master/src/main/java/alluxio/master/meta/MetaMasterInfo.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Metadata for an Alluxio standby masters.
+ * This class is not thread safe, so external locking is required.
+ */
+@NotThreadSafe
+public final class MetaMasterInfo {
+  private static final Logger LOG = LoggerFactory.getLogger(MetaMasterInfo.class);
+
+  /**
+   * Master's hostname.
+   */
+  private final String mHostname;
+  /**
+   * The id of the master.
+   */
+  private final long mId;
+  /**
+   * Start time of the master in ms.
+   */
+  private final long mStartTimeMs;
+  /**
+   * Master's last updated time in ms.
+   */
+  private long mLastUpdatedTimeMs;
+  /**
+   * If true, the master is considered registered.
+   */
+  private boolean mIsRegistered;
+
+  /**
+   * Creates a new instance of {@link MetaMasterInfo}.
+   *
+   * @param id       the master id to use
+   * @param hostname the master hostname to use
+   */
+  public MetaMasterInfo(long id, String hostname) {
+    mHostname = Preconditions.checkNotNull(hostname, "hostname");
+    mId = id;
+    mStartTimeMs = System.currentTimeMillis();
```

Comment:
This is more like the time the master registered. With failover the other master could have started much earlier than this time.

Reply:
Yes! It will not show the actual start time.

MetaMasterInfo now only contains three values: ``` hostname, id, lastUpdatedTimeMs```. Those are the values that we need now. If other values are needed, we could add them later.

Thanks for pointing out that.